internal/providers/k8s_test.go

```diff
@@ -159,18 +163,101 @@ func TestK8sRestore(t *testing.T) {
 	config.Providers.K8s.Features = defaultFeatureConfig
 
 	system := system.NewMockSystem()
+	// Mock that containerd service does not exist (typical case after k8s-only install)
+	system.MockCommandReturn("systemctl list-unit-files containerd.service", []byte("0 unit files listed."), nil)
+
 	ck8s := NewK8s(system, config)
 	ck8s.Restore()
 
-	expectedDeleted := []string{".kube"}
+	expectedRemovedPaths := []string{path.Join(os.TempDir(), ".kube")}
 
-	if !reflect.DeepEqual(expectedDeleted, system.Deleted) {
-		t.Fatalf("expected: %v, got: %v", expectedDeleted, system.Deleted)
+	if !reflect.DeepEqual(expectedRemovedPaths, system.RemovedPaths) {
+		t.Fatalf("expected: %v, got: %v", expectedRemovedPaths, system.RemovedPaths)
 	}
 
 	expectedCommands := []string{
 		"snap remove k8s --purge",
 		"snap remove kubectl --purge",
+		"systemctl list-unit-files containerd.service",
+	}
+
+	slices.Sort(expectedCommands)
+	slices.Sort(system.ExecutedCommands)
```

Comment:
Why the sort here (and similar below)? It seems to me that the original order is important and should be tested.